include/openthread/srp_server.h

```diff
@@ -262,7 +320,10 @@ const char *otSrpServerHostGetFullName(const otSrpServerHost *aHost);
 const otIp6Address *otSrpServerHostGetAddresses(const otSrpServerHost *aHost, uint8_t *aAddressesNum);
 
 /**
- * This function returns the next service of given host.
+ * This function returns the next service (excluding any sub-type services) of given host.
+ *
+ * @note This function is being deprecated and will be removed. `otSrpServerHostFindNextService()` can be used
+ *       instead.
```

Comment:
I kept these as before to keep it compatible with any existing next layer code using these APIs (they behave as before) before the subtype support is added

For example, these are being used  in `ot-br-posix` in `advertising_proxy.cpp`. I wanted to avoid having multiple PRs dependencies (PRs in different repos tied to each other and need to be merged together) to help make it easier to review and merge them seperately.